bundles/org.eclipse.osgi/felix/src/org/apache/felix/resolver/Candidates.java

```diff
@@ -907,36 +907,24 @@ public ResolutionError prepare()
                             // matter if they come from the host or fragment,
                             // since we are completing replacing the declaring
                             // host and fragments with the wrapped host.
-                            CandidateSelector cands = m_candidateMap.get(r);
-                            ShadowList shadow;
-                            if (!(cands instanceof ShadowList))
-                            {
-                                shadow = ShadowList.createShadowList(cands);
-                                m_candidateMap.put(r, shadow);
-                                cands = shadow;
-                            }
-                            else
-                            {
-                                shadow = (ShadowList) cands;
-                            }
 
-                            // If the original capability is from a fragment, then
-                            // ask the ResolveContext to insert it and update the
-                            // shadow copy of the list accordingly.
-                            if (!origCap.getResource().equals(hostResource.getDeclaredResource()))
+                            if (origCap.getResource().equals(hostResource.getDeclaredResource()))
                             {
-                                shadow.insertHostedCapability(
-                                        m_session.getContext(),
-                                        (HostedCapability) c,
-                                        new SimpleHostedCapability(
-                                                hostResource.getDeclaredResource(),
-                                                origCap));
+								// If the original capability is from the host, then
```

Comment:
Felix codebase does not use tab.

Reply:
Is this important to Equinox (so should we always have 1:1 code) or can it maybe addressed if/when back ported to felix? Can we somehow enable using space for the felix parts in JDT?)